wrappers/services.go

```diff
@@ -130,7 +130,16 @@ func StartServices(apps []*snap.AppInfo, inter interacter) (err error) {
 		}(app)
 
 		if len(app.Sockets) == 0 && app.Timer == nil {
-			services = append(services, app.ServiceName())
+			// check if the service is disabled, if so don't start it up
+			// this could happen for example if the service was disabled in
+			// the install hook by snapctl
+			isEnabled, err := sysd.IsEnabled(app.ServiceName())
```

Comment:
why is this check here, instead of at the top of the loop, right after the check for `IsService`?

If there is a reason for it to be here, we also need to not start the sockets and timers of the app.

Reply:
Well, effectively since the above function is deferred, that gets run only after we exit, and so this check is effectively run right after the `IsService`. 

I wasn't quite sure about disabling the sockets and timers case as I'm not familiar with how systemd timers work with snap services.
* If a snap service has a timer, does it have both a systemd unit for the timer and the service? Should we disable both of those if the service was disabled by `snapctl`? Probably yes
* Similar question for the socket, are there separate systemd units for the socket and the service? Should both of those be disabled if the snap service was disabled by `snapctl`?
* Also what does it mean to just run `snapctl stop SERVICE_WITH_TIMER` without the `--disable` flag during the install hook? Do we simply not run/start the timer on install? 
* Same question for sockets, what should happen when a `snapctl stop SERVICE_WITH_SOCKET` is run during install hook?

Reply:
The right mindset here is to think about the high-level concept we're dealing with. Most of these questions refer to the implementation instead. That is, for example, should a snap application that has a timer run when we call _snapctl stop_ on that application? Yes, I think it should.. the request was to stop what is running right now, not to disable it. Should it run when we disable it? That doesn't sound right.